src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -624,12 +627,14 @@ def construct_payload(self):
         self.set_up_repo()
         if self.get_argument_max_inactive_revisions() is not None:
             safe_set(self.containerapp_def, "properties", "configuration", "maxInactiveRevisions", value=self.get_argument_max_inactive_revisions())
+        self.set_up_runtime()
 
     def validate_arguments(self):
         super().validate_arguments()
         validate_create(self.get_argument_registry_identity(), self.get_argument_registry_pass(), self.get_argument_registry_user(), self.get_argument_registry_server(), self.get_argument_no_wait(), self.get_argument_source(), self.get_argument_artifact(), self.get_argument_repo(), self.get_argument_yaml(), self.get_argument_environment_type())
         if self.get_argument_service_bindings() and len(self.get_argument_service_bindings()) > 1 and self.get_argument_customized_keys():
             raise InvalidArgumentValueError("--bind have multiple values, but --customized-keys only can be set when --bind is single.")
+        validate_runtime(self.get_argument_runtime(), self.get_argument_enable_java_metrics())
```

Comment:
Arguments can be gotten with `self.` , no need to input:
self.get_argument_runtime(), self.get_argument_enable_java_metrics()

Reply:
`validate_runtime` function is being defined in `_validators.py`, it's not an object method. So, we could not get `self` inside the implementation.